sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -157,6 +157,7 @@ public static DataSourceConfiguration create(
       checkNotNull(url, "url");
       checkNotNull(username, "username");
       checkNotNull(password, "password");
+      checkArgument(password.length() > 0, "password");
```

Comment:
Actually, you are right. With the change on the `getConnection()` method, `username` and `password` can be null.